Fix duplicate Access List Terraform guides

ptgott opened the pull request:
See #58607

There are two how-to guides to managing Access Lists with IaC. One includes Operator and Terraform provider instructions, and one only includes Terraform instructions. Since the Terraform-only guide gets very little traffic, combine it with the broader guide. To do so, move information from the Terraform-specific guide to the general "Access Lists with IaC" guide if it is absent in the latter. This also has the advantage of updating the Access Lists with IaC guide.

ptgott committed:
Fix duplicate Access List Terraform guides

See #58607

There are two how-to guides to managing Access Lists with IaC. One
includes Operator and Terraform provider instructions, and one only
includes Terraform instructions. Since the Terraform-only guide gets
very little traffic, combine it with the broader guide. To do so, move
information from the Terraform-specific guide to the general "Access
Lists with IaC" guide if it is absent in the latter. This also has the
advantage of updating the Access Lists with IaC guide.

flyinghermit commented on docs/pages/identity-governance/access-lists/terraform.mdx:
```diff
@@ -1,348 +0,0 @@
-
-For static Access Lists, the source of truth for the membership is external (provisioned by
-Terraform) so audit is not supported. The members should be reviewed at source, which are the
-Terraform data sources or manifests and this process has to be external to Teleport.
```
This explanation is important but is missing from the new page https://github.com/gravitational/teleport/pull/60844/files#diff-7a487cfdfa16c5c6ca5d0653373db92bfb2eb73edc840f109e7295c9a6efb1b6

flyinghermit commented on docs/pages/identity-governance/access-lists/terraform.mdx:
```diff
@@ -1,348 +0,0 @@
-
-For static Access Lists, the source of truth for the membership is external (provisioned by
-Terraform) so audit is not supported. The members should be reviewed at source, which are the
-Terraform data sources or manifests and this process has to be external to Teleport.
```
This explanation of static and default access list type is important but is missing from the new page https://github.com/gravitational/teleport/pull/60844/files#diff-7a487cfdfa16c5c6ca5d0653373db92bfb2eb73edc840f109e7295c9a6efb1b6

ptgott replied:
I've edited the introduction to the H2 about managing Access List membership with Terraform to be closer to the the structure of the original discussion, reworking some awkward sentences.

r0mant reviewed (changes requested):
This PR removes access lists terraform guide from Identity Governance section which they are a part of and adds this guide to Zero-Trust Access instead for some reason.

Since we're refactoring, let's please make sure that things are where they belong. Access lists stuff should be nested under Identity Governance.

It's also missing the content that @flyinghermit pointed out in his review.

ptgott committed:
Edit the Access Lists Terraform guide

- Move the guide to the Identity Governance section.
- Expand the introduction to static vs. default Access Lists.

ptgott committed:
Move Access Lists TF guide to Managing Resources

While some features described in the Manaing Resources section are not
available in the Zero Trust Access product, keeping all how-to guides
for managing resources with IaC tools in this section will make it
easier to find a more discoverable location for this content in the
future, and for current users looking at these guides to find similiar
instructions.

Add a link from the Access Lists guide in the Identity Governance
section to the Managing Resources guide to make it more discoverable.
